src/VisualStudio/Core/Def/Preview/PreviewUpdater.cs

```diff
@@ -66,17 +72,27 @@ private void UpdateBuffer(TextDocument document)
 
             if (_currentDocumentId != null)
             {
-                Contract.ThrowIfNull(_previewWorkspace, "We shouldn't have a current document if we don't have a workspace.");
-                var currentDocument = _previewWorkspace.CurrentSolution.GetRequiredTextDocument(_currentDocumentId);
+                Contract.ThrowIfNull(previewWorkspace, "We shouldn't have a current document if we don't have a workspace.");
+                var currentDocument = previewWorkspace.Target.CurrentSolution.GetRequiredTextDocument(_currentDocumentId);
                 var currentDocumentText = currentDocument.GetTextSynchronously(CancellationToken.None);
-                _previewWorkspace.CloseDocument(currentDocument, currentDocumentText);
+                previewWorkspace.Target.CloseDocument(currentDocument, currentDocumentText);
+            }
+
+            if (previewWorkspace is null)
+            {
+                var newWorkspace = PreviewDialogWorkspace.Create(document.Project.Solution);
+                if (Interlocked.CompareExchange(ref _previewWorkspace, newWorkspace, null) is not null)
+                    newWorkspace.Dispose();
             }
 
-            _previewWorkspace ??= new PreviewDialogWorkspace(document.Project.Solution);
+            RoslynDebug.AssertNotNull(_previewWorkspace);
+
+            using var previewWorkspace2 = _previewWorkspace.TryAddReference();
+            RoslynDebug.AssertNotNull(previewWorkspace2);
```

Comment:
RequireAddReference :)

Reply:
➡️ Part of b228912c0d883c2f39f5413ace7f46cbd9c279b4